guides/session.rst

```diff
@@ -1,76 +1,114 @@
+    to see which features are supported by each driver.
 
 Basic Browser Interaction
-~~~~~~~~~~~~~~~~~~~~~~~~~
+-------------------------
 
-After you've instantiated the ``$session`` object, you can control the actual
-browser emulator with it:
+The first thing to do to use your session is to open a page with it. Just
+after starting, the session is not on any page (in a real browser, you would
+on ``about:blank``), and calling any other action is likely to fail.
 
 .. code-block:: php
 
-    // open some page in browser:
     $session->visit('http://my_project.dev/some_page.php');
 
-    // get the current page URL:
-    echo $session->getCurrentUrl();
+.. note::
 
-    // get the response status code:
-    echo $session->getStatusCode();
+    Mink is primarily designed to be used for testing websites. To allow
+    covering error pages as well, ``Session::visit`` does not consider that
+    error status codes are invalid. It will not throw an exception in this
+    case. You will need to check whether the response was a success or an
+    error. It will only throw an exception when Mink cannot load the page
+    (network error, ...).
+
+Interacting with the Page
+-------------------------
+
+The session gives you access to the page through the ``Session::getPage``
+method. This allows you to :doc:`traverse it </guides/traversing-pages>` and
+:doc:`manipulate it </guides/manipulating-pages>`. The next chapters are
+covering the page API in depth.
 
-    // get page content:
-    echo $session->getPage()->getContent();
+Using the Browser History
+-------------------------
 
-    // open another page:
-    $session->visit('http://my_project.dev/second_page.php');
+The session gives you access to the browser history:
+
+.. code-block:: php
+
+    // get the current page URL:
+    echo $session->getCurrentUrl();
 
     // use history controls:
     $session->reload();
     $session->back();
     $session->forward();
 
-    // evaluate JS expression:
-    echo $session->evaluateScript(
-        "return 'something from browser';"
-    );
+Cookies Management
+------------------
 
-    // wait for n milliseconds or
-    // till JS expression becomes true:
-    $session->wait(
-        5000,
-        "$('.suggestions-results').children().length > 0"
-    );
+The session can manipulate cookies available in the browser.
+
+.. code-block:: php
+
+    // set cookie:
+    $session->setCookie('cookie name', 'value');
+
+    // get cookie:
+    echo $session->getCookie('cookie name');
+
+    // delete cookie:
+    $session->setCookie('cookie name', null);
 
 .. note::
 
-    Although Mink does its best on removing browser differences between different
-    browser emulators, it can't do much in some cases. See the :ref:`driver-feature-support`
-    to see which features are supported by each driver.
+    In browser controllers, the access to http-only cookies may be restricted
+    as they cannot be accessed in Javascript.
+
+Status Code Retrieval
```

Comment:
You should also add note (from next section) saying that status code retrieval is only possible in headless drivers.